src/components/body/header/Header.tsx

```diff
@@ -1,82 +1,79 @@
 import Button from "../../Button";
+import Language from "./Language";
 import Logo from "../../icons/Logo.svg";
-import Magnifier from "../../icons/blackMagnifier.svg";
 import {nameDecodedUser} from "../../../store/header/decodeJwtSlice";
-import {showStudentPage} from "../../../store/header/headerSlice";
-import {useEffect} from "react";
 
-const Header = () => {
+type HeaderProps = {
+  onLanguageSwitch: (arg: string) => void;
+};
+
+const Header = ({onLanguageSwitch}: HeaderProps) => {
   const intl = useIntl();
   const dispatch = useAppDispatch();
   const navigate = useNavigate();
-  const isDefaultPage = useAppSelector(state => state.value.isDefaultHeader);
-  const page = useAppSelector(state => state.value.page);
+  const isRegister = useAppSelector(state => state.value.isDefaultHeader);
   const decodeUserName = useAppSelector(state => state.userDecodedName.session.name);
   const loading = useAppSelector(state => state.login.loading);
+  const [language, setLanguage] = useState("en");
 
   useEffect(() => {
     dispatch(nameDecodedUser());
     if (decodeUserName) {
       navigate("");
       dispatch(showStudentPage());
+    } else {
+      dispatch(showDefaultPage());
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isDefaultPage, decodeUserName, loading]);
-
-  const userButtonText = page === "sectionPage" ? intl.formatMessage({id: "app.header.myStudio"})
-    : intl.formatMessage({id: "app.header.goStudy"});
+  }, [isRegister, decodeUserName, loading]);
 
   return (
     <div className="side-bar">
       <div className="header">
-        <Link to="/n" className="logo-name">
+        <Link to="/" className="logo-name">
           <img className="logo" src={Logo} alt="Logo"/>
           <h4 className="title-header">
             <FormattedMessage id="app.name"/>
           </h4>
-          {(page === "myPage" && isDefaultPage) && <Link to={"/myStudio"} className="my-studio">
-            <FormattedMessage id="app.studio"/>
-          </Link>}
         </Link>
-        <div className="search-button">
-          <Link to="/search" className="magnifier">
-            <img src={Magnifier} alt="search"/>
-          </Link>
-          <input
-            className="search" type="text"
-            placeholder={intl.formatMessage({id: "app.header.placeholder"})}
-          />
-          {isDefaultPage ?
-            <div className="user-item">
-              {page &&
-                <Link to="/" className="section-button">
-                  <Button
-                    buttonType="button"
-                    buttonText={userButtonText}
-                    className="user-button"
-                  />
-                </Link>}
-              <img src={Bell} alt="Bell" className="bell"/>
-              <Avatar/>
-            </div>
-            :
-            <Link to="/users/sign_in" className="login-link">
+        <div className="header-buttons">
+          {isRegister ? <Avatar
```

Comment:
it is better 

{isRegister && ,,,,}
{!isRegister && ,,,}

instead of ternary operstor as it difficult to inspect in the code